### DISCUSSION TITLE
Release assets not appearing

### DISCUSSION THREAD

**Discussion by andymantell** [community/community#25347] [Actions]
<p>Hi,</p>
<p>I have a Github action setup to create a release and upload binaries to it, however I am experiencing some odd behaviour. If I view the release on the following URL I do not see any assets apart from the source code:</p>
<p><code>https://github.com/andymantell/govuk-frontend-diff/releases/tag/v0.1-alpha-4</code></p>
<p>This is the URL that I can reach by clicking through the releases section in the Github UI. However, when the action has run I also receive an email from github saying that a new release has been created at:</p>
<p><code>https://github.com/andymantell/govuk-frontend-diff/releases/tag/refs/tags/v0.1-alpha-4</code></p>
<p>This one has all the artifacts that I am expecting to see.</p>
<p>As far as I can see, they are one and the same release. But they appear to have different URLs (One has an additional <code>refs/tags</code> section in it.</p>
<p>I don’t quite understand what’s going on here. Can anyone shed any light on this?</p>
<p>Thanks,</p>
<p>Andy</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/andymantell">@andymantell</a>,</p>
<p>Glad to see you in Github Community Forum!<br>
To add assets to an existing release, please use ‘edit’ instead of ‘create’ for command ‘hub release’.<br>
Option ‘create’ will create a GitHub release for the specified TAG name. If git tag TAG does not exist, it will be created at TARGET (default: current branch).<br>
Please fix code as below:</p>
<pre><code class="lang-auto">      - name: Attach binaries to release
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        run: |
          set -x
          assets=()
          for asset in ./bin/*; do
            assets+=("-a" "$asset")
          done
          githubref=${{ github.ref }}   # get the github.ref value
          hub release edit "${assets[@]}" -m "${githubref##*/}" "${githubref##*/}"   # specify tag name not the github.ref here.
</code></pre>
<p>It works on my side. Please check my workflow <a href="https://github.com/weide-zhou/govuk-frontend-diff/runs/775092143?check_suite_focus=true#step:7:1" rel="nofollow noopener">here</a>.</p>
<p>Thanks</p>

**Comment by andymantell** [comment]
<p>Thanks <a class="mention" href="/weide-zhou">@weide-zhou</a>! I have made the changes you suggest and it’s now working perfectly.</p>
<p>Thankyou very much for the quick response, very much appreciated.</p>
<aside class="onebox githubcommit">
  <header class="source">
      <a href="https://github.com/andymantell/govuk-frontend-diff/commit/f3c246c556a69a7daa529d9d11ef1ebe0240a0e5" target="_blank" rel="nofollow noopener">github.com/andymantell/govuk-frontend-diff</a>
  </header>
  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
    <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M10.86 7c-.45-1.72-2-3-3.86-3-1.86 0-3.41 1.28-3.86 3H0v2h3.14c.45 1.72 2 3 3.86 3 1.86 0 3.41-1.28 3.86-3H14V7h-3.14zM7 10.2c-1.22 0-2.2-.98-2.2-2.2 0-1.22.98-2.2 2.2-2.2 1.22 0 2.2.98 2.2 2.2 0 1.22-.98 2.2-2.2 2.2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/andymantell/govuk-frontend-diff/commit/f3c246c556a69a7daa529d9d11ef1ebe0240a0e5" target="_blank" rel="nofollow noopener">Fix release.yml to make sure binaries get associated with the releases</a>
    </h4>

    <div class="github-info">
      <div class="date">
        committed <span class="discourse-local-date" data-format="ll" data-date="2020-06-16" data-time="07:34:29" data-timezone="UTC">07:34AM - 16 Jun 20 UTC</span>
      </div>

      <div class="user">
        <a href="https://github.com/andymantell" target="_blank" rel="nofollow noopener">
          <img alt="andymantell" src="https://user-images.githubusercontent.com/134642/181090559-f9d333a3-8cfb-45e5-bd24-1214c7d9b07a.jpeg" class="onebox-avatar-inline" width="20" height="20">
          andymantell
        </a>
        
      </div>

      <div class="lines" title="changed 1 files with 2 additions and 1 deletions">
        <a href="https://github.com/andymantell/govuk-frontend-diff/commit/f3c246c556a69a7daa529d9d11ef1ebe0240a0e5" target="_blank" rel="nofollow noopener">
          <span class="added">+2</span>
          <span class="removed">-1</span>
        </a>
      </div>
    </div>

  </div>
</div>


  <div class="github-row">
    <pre class="github-content" style="white-space: normal;">See https://github.community/t/release-assets-not-appearing/118148
Co-authored-by: weide-zhou &lt;weide-zhou@outlook.com&gt;</pre>
  </div>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>
